### PR TITLE
Features/split pane

### PR DESCRIPTION
**Pull request by agtenr**
## Type of change

- [ ] Bugfix
- [x] Component improvement
- [x] New component
- [ ] New package
- [ ] Other

## Description
 - New component: SplitPane
 - Update component: Add disable option to useDrag hook

### CODE REVIEW

File: packages/dw-react-controls/src/components/SplitPane/SplitPane.styles.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	separatorRight: "dw-SplitPane-separatorRight"
+};
+
+export const getStyles = (props: ISplitPaneStyleProps): ISplitPaneStyles => {
+	const { className, fixed, isDragging } = props;
+	const classNames = GlobalClassNames;
+
+	return {
+		container: [
+			className,
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/SplitPane/SplitPane.styles.ts:18]: shouldn't `className` be last, to ensure css specificity when overriding?

File: packages/dw-react-controls/src/components/SplitPane/SplitPane.base.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+			leftPane.current.style.width = `${leftPaneNewWidth}%`;
+			rightPane.current.style.width = `${rightPaneNewWidth}%`;
+		}
+	}
+
+	// Styles
+	const classNames = getClassNames(styles, { className, fixed, isDragging });
+
+	return (
+		<div>
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/SplitPane/SplitPane.base.tsx:63]: is the extra `<div>` needed?

File: packages/dw-react-controls/src/components/SplitPane/SplitPane.types.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import { IStyle } from "office-ui-fabric-react/lib/Styling";
+import { IStyleFunctionOrObject } from "office-ui-fabric-react/lib/Utilities";
+
+export interface ISplitPaneProps {
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/SplitPane/SplitPane.types.ts:4]: there's something wrong with <kbd>tab</kbd> and <kbd>space</kbd> formatting here :)

File: packages/dw-react-controls/src/components/SplitPane/stories/SplitPane.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import { ArgsTable, Meta, Story, Canvas } from "@storybook/addon-docs";
+import { SplitPane } from "../SplitPane"
+import * as stories from "./SplitPane.stories";
+
+<Meta title="Components/SplitPane" component={SplitPane} />
+
+# SplitPane
+
+The	 `SplitPane` can be used to split the screen into two parts.
```
Comment (NickSevens) [inline packages/dw-react-controls/src/components/SplitPane/stories/SplitPane.stories.mdx:9]: weird spacing here

File: packages/dw-react-utils/src/hooks/useDrag.ts [hunk 5]
```diff
@@ -10,19 +10,22 @@ import { useWindowEvent } from "..";
...
  * @returns isDragging - Boolean indicating if the dragging is ongoing
  * @returns positionX - The current X position of the drag
  * @returns positionY - The current Y position of the drag
  */
 export const useDrag = (
   element: HTMLElement | undefined,
   direction?: "vertical" | "horizontal",
-  onDragEnd?:(positionX: number, positionY: number) => void
+  onDragEnd?: (positionX: number, positionY: number) => void,
+	disabled?: boolean
```
Comment (NickSevens) [inline packages/dw-react-utils/src/hooks/useDrag.ts:22]: weird spacing

File: packages/dw-react-utils/src/stories/hooks/useDrag.stories.tsx [hunk 6]
```diff
@@ -62,6 +63,35 @@ export const Horizontal: Story = () => {
   )
 }
 
+export const Disable: Story = () => {
+  const dragRef = React.useRef() as React.MutableRefObject<HTMLDivElement>;
+  const [dragElement, setDragElement] = React.useState<HTMLElement>();
+	const [disabled, setIsDisabled] = React.useState(false);
+  const { isDragging, positionX, positionY } = useDrag(dragElement, "horizontal", undefined, disabled);
+
+	const onChange = (e: React.FormEvent<HTMLInputElement>) => {
```
Comment (NickSevens) [inline packages/dw-react-utils/src/stories/hooks/useDrag.stories.tsx:72]: another weird spacing (i'm beginning to suspect something wrong in GitHub rendering :p)

File: packages/dw-react-controls/src/components/SplitPane/SplitPane.styles.ts (hunk 1 above)
Comment (agtenr) [reply packages/dw-react-controls/src/components/SplitPane/SplitPane.styles.ts]: True!

File: packages/dw-react-controls/src/components/SplitPane/SplitPane.base.tsx (hunk 2 above)
Comment (agtenr) [reply packages/dw-react-controls/src/components/SplitPane/SplitPane.base.tsx]: Nope :)

**Comment by NickSevens** [comment]
Looks great to me! 👍🏻

**Pull request merged**